src/librustdoc/html/render/print_item.rs

```diff
@@ -1416,36 +1417,83 @@ fn item_enum(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, e: &clean::
             it.name.unwrap(),
             e.generics.print(cx),
         );
+
         render_enum_fields(
             w,
             cx,
             Some(&e.generics),
-            e.variants(),
+            &e.variants,
             count_variants,
             e.has_stripped_entries(),
             it.is_non_exhaustive(),
+            it.def_id().unwrap(),
         );
     });
 
     write!(w, "{}", document(cx, it, None, HeadingOffset::H2));
 
     if count_variants != 0 {
-        item_variants(w, cx, it, e.variants());
+        item_variants(w, cx, it, &e.variants);
     }
     let def_id = it.item_id.expect_def_id();
     write!(w, "{}", render_assoc_items(cx, it, def_id, AssocItemRender::All));
     write!(w, "{}", document_type_layout(cx, def_id));
 }
 
-fn render_enum_fields<'a>(
+/// It'll return true if all variants are C-like variants and if at least one of them has a value
+/// set.
+fn should_show_c_like_variants_value(
```

Comment:
```suggestion
fn should_show_enum_discriminant(
```

I prefer the term “(enum) discriminant” over “variant value”. Minor nitpick: The part `c_like_` of the fn name is a  bit misleading since this function does in fact handle all sorts of enum variants just fine, it just so happens that it returns `false` for non-C-like variants.

Reply:
Also, the PR title could be updated to say “enum discriminant” instead of “enum variant value”.